### ISSUE TITLE
UFIXIT - HTMLEncoded HTML markup inside <code> block becomes unencoded after <img> alt text is fixed

### ISSUE THREAD

**Issue by tsummers77** [ucfopen/UDOIT#576]
An assignment has a ```<code>``` block containing HTML-Encoded HTML markup, and also an ```<img>``` tag that is missing alt text/decorative attribute:

```html
<pre>
    <code>
    &lt;strong&gt;weak&lt;/strong&gt;
    &lt;strong&gt;weaker&lt;/strong&gt;
    </code>
</pre>

<p>
    <img src="https://a.b.c/img.jpg" width="320" height="320" />
</p>
```

If UFIXIT is used to supply alt text / decorative marker, the HTML-Encoded markup becomes decoded and is then improperly rendered:

```html
<pre>
    <code>
        <strong>weak</strong>
        <strong>weaker</strong>
    </code>
</pre>

<p>
    <img src="https://a.b.c/img.jpg" alt="" width="320" height="320" data-decorative="true" />
</p>
```

**Comment by bagofarms** [comment]
Verified that this is affecting my installation, too.

**Comment by tsummers77** [comment]
Taking a very quick look at [https://github.com/ucfopen/UDOIT/blob/master/lib/Ufixit.php](url), it seems the problem lies with how ```$annoying_entities``` are replaced with ```$replacement_entities``` within the ```replaceContents()``` function. Perhaps using CDATA for replacements would fix the issue:

For example, replace ```&lt;``` with ```<![CDATA[&lt;]]>``` instead of with ```<```

This of course eats up more memory but as long as HTMLMinify recognizes CDATA entities properly when it minifies it might be a good solution.
